src/Runner.hpp

```diff
@@ -55,27 +55,39 @@ class CabanaPFRunner {
         return result;
     }
 
-    void timestep(int count) {
+    // Do a certain number of timesteps:
+    void run(const int timesteps) {
         if (!have_initialized) {
             initialize();
             have_initialized = true;
         }
-        for (int i = 0; i < count; i++) {
-            pre_step();
+        for (int i = 0; i < timesteps; i++) {
             step();
-            post_step();
             timesteps_done++;
         }
-        if (timesteps_done == timesteps)
-            finalize();
     }
 
-    // Generally, you inherit from this class and implement one or more of these:
+    void run_for_time(const double time) {
```

Comment:
Is this needed? Isn't this the same as `run_until_time` without previous steps?

Reply:
run_for_time is "go forward by 5 seconds" while run_until_time is "go forward until you get to a certain time".  So if we're already at t=2s and call run_for_time(5), we'll be at t=7s, while calling run_until_time(5) will hit t=5s.

I thought that it would be useful to offer both: this lets us offer the user all 4 combinations of (working in timesteps or time) x (specifying how far to go or where to wind up).  I can remove it though if you don't think it's necessary.

Reply:
I'm fine either way, as long as the naming is consistent. Could use simpler names by using `enable_if` to enforce the correct types and then use `run()` and `run_until()` with separate int/double overloads

Reply:
I would prefer not to do overloads, I think that could be confusing. But I do think consistent terminology would be good. That would mean changing `run` to `run_for_steps`. That way the four options for moving forward in time are clearly specified.

Reply:
Let's do the simpler version then 👍